dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java

```diff
@@ -638,11 +586,19 @@ else if(PERSON_CONTEXTCLASS.equals(contextClass))
                                     + "Please restore this Person object using the SITE AIP, or recreate it.");
                         }
 
-                        //assign permissions to person on this object
-                        assignPermissions(context, dso, person, permsElement);
+                        //assign person to the policy
+                        rp.setEPerson(person);
                     }//end if Person
-                    else
+                    else {
                         log.error("Unrecognized CONTEXTCLASS:  " + contextClass);
+                    }
+                    
+                    //set permissions on policy and add to object
+                    rp.setAction(parsePermissions(permsElement));
+                	policies.add(rp);
+                	assignPermissions(context, dso, policies);
+                } catch (NullPointerException ex) {
+                    java.util.logging.Logger.getLogger(METSRightsCrosswalk.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
Another silly question: Why are we catching a NullPointerException here? Is there a check for possible null value that is missing?

It'd be good to have an explanation. If it is necessary, I'd recommend we comment this, as usually we don't see catches for NPEs (we just try to avoid null values).

It's also worth noting that you don't need the complex logger line of code. It could be simplified to just `log.error([explanation], ex);`.  There's already a global reference for a logger at [line 64](https://github.com/Georgetown-University-Libraries/DSpace/blob/05d85843e90d23ace5cad231e05ccb5fa3096eeb/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java#L64)